File: src/javascript/compress_ew2.js
const assert = require("assert");

//     v 70 chars, no regex /\/\/\:
compress=(s,r="",c=1)=>[...s].map((v,i)=>v==s[i+1]?c++:[r+=c+v,c=1]).pop()[0]

assert.equal(compress("AAABBAAC"), "3A2B2A1C");

File: src/python/compress_ew2.py
def compress(s='', r='', c=1):
    for p, v in zip(s, s[1:] + '0'):
        [r,c]=([r+str(c)+p,1],[r,c+1])[p==v]
    return r


assert compress("AAABBAAC") == "3A2B2A1C"
